### PR TITLE
Solves issue 220

### PR DESCRIPTION
**Pull request by TG1999**
#220 solved
This patch includes:
- [x] A relevant test
- [ ] A relevant documentation update

### CODE REVIEW

File: lib/tokeniser.js [hunk 1]
```diff
@@ -251,3 +251,6 @@ class WebIDLParseError extends Error {
     this.tokens = tokens;
   }
 }
+module.exports={
```
Comment (saschanaz) [inline lib/tokeniser.js]: This is old CommonJS way to export things, here instead we want to use more modern way: just add `export` before `class WebIDLParseError` 👍

File: index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export { parse } from "./lib/webidl2.js";
 export { write } from "./lib/writer.js";
 export { validate } from "./lib/validator.js";
+export {Err} from "./lib/tokeniser.js";
```
Comment (saschanaz) [inline index.js]: This is good, but I would prefer the original name `WebIDLParseError`.

Comment (saschanaz) [reply index.js]: Make sure we get spaces between braces!

File: index.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
 export { parse } from "./lib/webidl2.js";
 export { write } from "./lib/writer.js";
 export { validate } from "./lib/validator.js";
+export {WebIDLParseError} from "./lib/tokeniser.js";
```
Comment (saschanaz) [inline index.js]:
```suggestion
export { WebIDLParseError } from "./lib/tokeniser.js";
```

For code format consistency 😁 (You can click "Commit suggestion" to automatically accept my suggestion)

File: lib/tokeniser.js [hunk 4]
```diff
@@ -251,3 +251,4 @@ class WebIDLParseError extends Error {
     this.tokens = tokens;
   }
 }
+
```
Comment (saschanaz) [inline lib/tokeniser.js]:
```suggestion
```

Please remove redundant newline here!

File: index.js (hunk 2 above)
Comment (TG1999) [reply index.js]: Okay I am adding spaces now it is also passing all the tests, so I think it can be merged after that change :)

**Comment by TG1999** [comment]
@saschanaz all the changes requested in review are done so I think you can merge it now :)

**Comment by saschanaz** [comment]
Code now looks great, but we have slightly more longer work here: Adding a test!

We want to add a test in [test/invalid.js](https://github.com/w3c/webidl2.js/blob/gh-pages/test/invalid.js), where we want to make sure `err instanceof WebIDLParseError` (which is our original purpose) works as intended.

At line 131, we want to add a new line `it("allows `instanceof WebIDLParseError`", () => {` and check it works as described. `parse("throwerror")` will throw the error we exported just now, so you'll want to try-catch that error and `expect(err).toBeInstanceOf(WebIDLParseError)`.

**Comment by TG1999** [comment]
Okay on it

**Comment by TG1999** [comment]
@saschanaz any suggestions why this is failing ? Somngthing regarding whitespace error is given at 132:1 I am not able to understand it

**Comment by saschanaz** [comment]
I don't see a trailing whitespace there... but try `npm run lint -- --fix` and it will automatically try fixing the error, if you have a local clone of this repo.

File: test/invalid.js [hunk 5]
```diff
@@ -129,6 +129,16 @@ describe("Error object structure", () => {
     const { level } = validation[0];
     expect(level).toBe("warning");
   });
+
```
Comment (saschanaz) [inline test/invalid.js]:
```suggestion

```

Oh, I got it, we do have trailing spaces here.

**Comment by TG1999** [comment]
Okay if I will fix at my local too, it will still give an error here, so how will we pass the tests here ?

File: test/invalid.js [hunk 6]
```diff
@@ -129,6 +129,16 @@ describe("Error object structure", () => {
...
+  
+  it("allows `instanceof WebIDLParseError`", () => {
+    try {
+      parse("throwerror");
+      throw new Error("Shouldn't reach here");
+    } catch ({err}) {
+      expect(err).toBeInstanceOf(WebIDLParseError);
+    }
+  });
+
```
Comment (saschanaz) [inline test/invalid.js]:
```suggestion
```

This line looks redundant.

**Comment by TG1999** [comment]
Sorry I closed it by mistake, just making the changes. :)

**Comment by TG1999** [comment]
Hey @saschanaz please can you have  look again :sweat_smile:

File: test/invalid.js [hunk 7]
```diff
@@ -129,6 +129,14 @@ describe("Error object structure", () => {
     const { level } = validation[0];
     expect(level).toBe("warning");
   });
+  it("allows `instanceof WebIDLParseError`", () => {
```
Comment (saschanaz) [inline test/invalid.js]:
```suggestion

  it("allows `instanceof WebIDLParseError`", () => {
```

Oops, we accidentally removed the newline before this line 😲

File: test/invalid.js [hunk 8]
```diff
@@ -129,6 +129,14 @@ describe("Error object structure", () => {
...
+    try {
+      parse("throwerror");
+      throw new Error("Shouldn't reach here");
+    } catch ({err}) {
```
Comment (saschanaz) [inline test/invalid.js]: `{err}` accesses `err` field of the error object, I think we just want `err` to get the error object itself.

Comment (TG1999) [reply test/invalid.js]: Okay I have made the changes and they have passed. Thanks for your help :smile:

**Pull request merged**